Python/flowgraph.c

```diff
@@ -1586,6 +1586,54 @@ optimize_cfg(cfg_builder *g, PyObject *consts, PyObject *const_cache)
     return SUCCESS;
 }
 
+static void
+make_super_instruction(cfg_instr *inst1, cfg_instr *inst2, int super_op)
+{
+    /* Skip if instructions are on different lines */
+    if (inst1->i_loc.lineno != inst2->i_loc.lineno) {
+        return;
```

Comment:
I saw a comment from @brandtbucher that this requirement eliminated most super-instructions (https://github.com/faster-cpython/ideas/issues/585#issuecomment-1555119860). And yet your pyperformance results on this PR are quite good. I wonder how these two things are both true.

Reply:
For instructions without observable side effects (e.g. `LOAD_FAST`, or maybe `POP_TOP`), can't we still do a super instruction and keep the `NOP` for the extra lineno?

Reply:
> I saw a comment

Oh, I realize now that this comment was specific to `STORE_FAST__LOAD_FAST`.

Reply:
> I wonder how these two things are both true.

Oh, I guess the most likely answer is in https://github.com/faster-cpython/ideas/issues/585#issuecomment-1573433691 :

> it appears that the reduction in code size more than compensates.

Reply:
That is just my hypothesis.

The aim is to remove superinstructions that cross block and line boundaries, as they are problematic.
I was just aiming to not slow things down by removing them. A speedup was just a pleasant surprise.
Even if the speedup is mainly from layout changes, or cache effects, the main motivation of this PR remains valid.